src/com/goide/intentions/GoMoveToStructInitializationIntention.java

```diff
@@ -0,0 +1,283 @@
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiWhiteSpace;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.IncorrectOperationException;
+import com.intellij.util.ObjectUtils;
+import com.intellij.util.containers.ContainerUtil;
+import com.intellij.util.containers.MultiMap;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class GoMoveToStructInitializationIntention extends BaseElementAtCaretIntentionAction {
+  public static final String NAME = "Move field assignment to struct initialization";
+
+  public GoMoveToStructInitializationIntention() {
+    setText(NAME);
+  }
+
+  @Nls
+  @NotNull
+  @Override
+  public String getFamilyName() {
+    return NAME;
+  }
+
+  @Override
+  public boolean isAvailable(@NotNull Project project, Editor editor, @NotNull PsiElement element) {
+    if (!element.isValid() || !element.isWritable()) return false;
+    GoAssignmentStatement assignment = getValidAssignmentParent(element);
+    GoReferenceExpression selectedFieldReference = assignment != null ? getFieldReference(element, assignment) : null;
+    return selectedFieldReference != null && !getUninitializedSingleFieldReferences(assignment, selectedFieldReference).isEmpty();
+  }
+
+  @Override
+  public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element) throws IncorrectOperationException {
+    if (!element.isValid() || !element.isWritable()) return;
+    GoAssignmentStatement assignment = getValidAssignmentParent(element);
+    if (assignment == null) return;
+    GoReferenceExpression selectedFieldReference = getFieldReference(element, assignment);
+    if (selectedFieldReference == null) return;
+    GoCompositeLit structLiteral = getStructLiteralByReference(selectedFieldReference, assignment);
+    if (structLiteral == null) return;
+    moveFields(getUninitializedSingleFieldReferences(assignment, selectedFieldReference), structLiteral, project);
+  }
+
+  @Nullable
+  private static GoAssignmentStatement getValidAssignmentParent(@Nullable PsiElement element) {
+    return ObjectUtils.nullizeByCondition(PsiTreeUtil.getNonStrictParentOfType(element, GoAssignmentStatement.class),
+                                          assignment -> assignment == null ||
+                                                        !assignment.isValid() ||
+                                                        !isAssignmentEqualsSized(assignment));
+  }
+
+  private static boolean isAssignmentEqualsSized(@NotNull GoAssignmentStatement assignment) {
+    List<GoExpression> expressions = assignment.getExpressionList();
+    return assignment.getLeftHandExprList().getExpressionList().size() == expressions.size() &&
+           expressions.stream().allMatch(e -> GoInspectionUtil.getExpressionResultCount(e) == 1);
```

Comment:
I agree, an intention is a mini-refactoring, it's not supposed to fix anything. I mean if there is a red code that it's ok to keep is red after intention action applying.